src/main/java/net/minecraftforge/items/IItemHandler.java

```diff
@@ -61,12 +62,13 @@
     /**
      * <p>
      * Inserts an ItemStack into the given slot and return the remainder.
-     * The ItemStack <em>should not</em> be modified in this function!
```

Comment:
This sounds like a very annoying contract change, is there a reason for it?

Reply:
This is actually a documentation update, as the current documentation has been wrong for a long time. This comes from an old PR of mine https://github.com/TheCurle/MinecraftForge/pull/17.

Reply:
![image](https://user-images.githubusercontent.com/13494793/144524051-d99a9c95-b576-441b-8b2b-ba91abb53085.png)
As you can see in ItemStackHandler, if `reachedLimit` is false the stack is directly set.

Reply:
Reverted the javadoc change, and fixed the bug in ItemStackHandler instead.